zilliqa/src/consensus.rs

```diff
@@ -496,11 +497,25 @@ impl Consensus {
             .unwrap()
     }
 
-    // This function is called when we suspect that we are out of sync with the network/need to catchup
+    /// This function is called when we suspect that we are out of sync with the network/need to catchup.
+    /// We ask peers for their chain above our head, if the network is syncronised there should be nothing
+    /// to return.
```

Comment:
open to debate on this solution - it sort of assumes that validators are the only people who exist on the network which seems incorrect/bad

Reply:
Yes, I agree with this, but it is def a TODO which will have to be addressed at some point.